src/server/api/routers/cart.ts

```diff
@@ -65,4 +65,23 @@ export const cartRouter = createTRPCRouter({
         where: { id: input.cartId },
       });
     }),
+  getCart: publicProcedure
+    .input(z.object({ cartId: z.string() }))
+    .query(async ({ ctx, input }) => {
+      const { cartId } = input;
+      const cart = await ctx.db.cart.findUnique({ where: { id: cartId } });
+      if (!cart) {
+        throw new TRPCError({
+          message: "Cart not found",
+          code: "INTERNAL_SERVER_ERROR",
+        });
+      }
+      const cartItems = await ctx.db.cartItem.findMany({ where: { cartId } });
```

Comment:
I don't think you need to find the cart items since the car items should be apart of the cart based on cartId so you should just be able to return cart if there is a cart!